packages/core-transaction-pool/lib/guard.js

```diff
@@ -117,15 +118,22 @@ module.exports = class TransactionGuard {
    * Transforms and filters incomming transactions.
    * It skips duplicates and not valid crypto transactions
    * It skips blocked senders
+   * It skips not matching fees, but still adds them to broadcast mode
    * @param  {Array} transactions
    * @return {void}
    */
   __transformAndFilterTransactions (transactions) {
     this.transactions = []
 
     transactions.forEach(transaction => {
-      const exists = this.pool.transactionExists(transaction.id)
+      if (!dynamicFeeMatch(transactions)) {
+        this.broadcast.push(transaction)
+        // TODO: currently no id is returned to the sender so he doesnt know what is with this id?
```

Comment:
Why is no fee returned to the sender, shouldn't the ID be in the broadcast array?

Maybe we should also push the transaction as an error with the message `Peer rejected the transaction because of not meeting the minimum accepted fee.` so the user knows it has been broadcasted *but* also gets an indication of why it was not accepted since *rejection* technically is an error.